x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/evaluator/predicate/operator/comparison/InsensitiveEquals.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.esql.evaluator.predicate.operator.comparison;
+
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.automaton.ByteRunAutomaton;
+import org.elasticsearch.compute.ann.Evaluator;
+import org.elasticsearch.compute.ann.Fixed;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.tree.NodeInfo;
+import org.elasticsearch.xpack.ql.tree.Source;
+import org.elasticsearch.xpack.ql.type.DataTypes;
+
+import java.time.ZoneId;
+
+public class InsensitiveEquals extends InsensitiveBinaryComparison {
+
+    public InsensitiveEquals(Source source, Expression left, Expression right, ZoneId zoneId) {
+        super(source, left, right, zoneId);
+    }
+
+    @Override
+    public InsensitiveBinaryComparison reverse() {
+        return this;
+    }
+
+    @Override
+    protected NodeInfo<InsensitiveEquals> info() {
+        return NodeInfo.create(this, InsensitiveEquals::new, left(), right(), zoneId());
+    }
+
+    @Override
+    protected InsensitiveEquals replaceChildren(Expression newLeft, Expression newRight) {
+        return new InsensitiveEquals(source(), newLeft, newRight, zoneId());
+    }
+
+    @Evaluator(extraName = "Ints")
```

Comment:
Yeah, I don't have a strong opinion on this. I just implemented it as it was described in the initial issue https://github.com/elastic/elasticsearch/issues/103599.

And yes, I think I can simplify it an in case use Equals implementation

Reply:
@costin do you think `=~` should allow comparisons with non-string types? I think maybe no?

Reply:
+1 on falling back to the regular Equals operators for non string params.